subsys/net/lib/coap/coap_client.c

```diff
@@ -896,7 +907,9 @@ void coap_client_recv(void *coap_cl, void *a, void *b)
 					continue;
 				}
 
-				ret = handle_response(clients[i], &response);
+				response_truncated = ret > MAX_COAP_MSG_LEN;
```

Comment:
I think it'd make sense if `recv_response()` accepted additional `bool *truncated` parameter, and fill out this information.